spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractMessageChannelBinder.java

```diff
@@ -179,6 +180,9 @@ protected void afterUnbindProducer(String destination, P producerProperties) {
 			bridge.setBeanName("bridge." + name);
 			consumerEndpoint = createConsumerEndpoint(name, group, destination, properties);
 			consumerEndpoint.setOutputChannel(bridge);
+			if(consumerEndpoint instanceof IntegrationObjectSupport) {
+				((IntegrationObjectSupport) consumerEndpoint).afterPropertiesSet();
```

Comment:
Also, we need to do the same for the producer handler and documented the contract in the Javadoc.